Hwmv2 various soc name sync

tejlmand opened the pull request:
Align SoC names in soc.yml files with Kconfig SOC settings.

The will allow to enable the CI check in: #69309

tejlmand committed:
soc: ti: define SOC name in Kconfig

Add Kconfig SoC name am6234 which matches the SoC name in soc.yml.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

tejlmand committed:
soc: mec: align SoC names defined in soc.yml to Kconfig SOC setting

This commit aligns the SoC names for mec SoCs defined in soc.yml to
Kconfig SOC setting.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

tejlmand committed:
soc: arm: align SoC names defined in soc.yml to Kconfig SOC setting

This commit aligns the SoC names for arm an547 SoCs defined in soc.yml
to Kconfig SOC setting.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

tejlmand committed:
soc: broadcom: align SoC names defined in soc.yml to Kconfig SOC setting

This commit aligns the SoC names for bcm58402 SoCs defined in soc.yml
to Kconfig SOC setting.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

tejlmand commented:
@danieldegrasse you introduced `mk82f215 ` here: aeef9e5f4adc13a3bfce2ea56ba7312bc42cafd9
https://github.com/zephyrproject-rtos/zephyr/blob/d6cb9c967227b05e59a3f8e5a619fa695c174594/soc/nxp/kinetis/soc.yml#L24

However, this SoC has no Kconfig implementation and no board seems to be using this SoC.
Can it be removed or what is the intention with this SoC ?

danieldegrasse commented:
> However, this SoC has no Kconfig implementation and no board seems to be using this SoC.
Can it be removed or what is the intention with this SoC ?

Thanks for catching this- I think this was a typo. The correct SOC here should be `mk80f25615`, which is present in the Kconfig.soc file but not in soc.yml

tejlmand committed:
soc: nxp: fix typo in SoC name

Follow-up: #68778

There is a typo in nxp/kinetics/soc.yml.
The SoC `mk82f215` was added, however the correct SoC is `mk80f25615`,
which this commit corrects.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

tejlmand commented:
> Thanks for catching this- I think this was a typo. The correct SOC here should be `mk80f25615`, which is present in the Kconfig.soc file but not in soc.yml

@danieldegrasse I have added a commit in this PR fixing this typo. Please take a look.
